lib/get-soundcloud-url.js

```diff
@@ -0,0 +1,22 @@
+const Request = require('sdk/request').Request;
+const clientId = require('../package.json').config['SOUNDCLOUD_CLIENT_ID'];
+
+module.exports = getSoundcloudUrl;
+
+function getSoundcloudUrl(videoUrl, cb) {
+  Request({
+    url: `https://api.soundcloud.com/resolve.json?client_id=${clientId}&url=${videoUrl}`,
+    onComplete: function (resp) {
+      if (!resp.json) {
+        cb('Could not connect to soundcloud.com');
+        return;
+      }
+
+      if (resp.json.kind !== 'track' || !resp.json.streamable) {
+        cb('not track or not streamable');
```

Comment:
For telling the user what went wrong, might be useful to split this error to distinguish between the 'not track' and 'not streamable' cases? Maybe a good first bug?